src/utils/common/common.tsx

```diff
@@ -9,6 +10,28 @@ import { fromB64Url } from '../encodings';
 export function formatDate(epochMs: number): string {
   return new Date(epochMs).toISOString().split('T')[0];
 }
+
+export function formatLeaseDurationFromEndTimestamp(
```

Comment:
Odd to see this as a utility function when it's essentially a React.FC. I'd rename it with a capitalized name (e.g., LeaseDuration) and move it into its own component file.